frontend/dockerfile/builder/build.go

```diff
@@ -874,6 +874,18 @@ func contextByName(ctx context.Context, c client.Client, name string, platform *
 			st = &httpst
 		}
 		return st, nil, nil, nil
+	case "oci-layout":
+		ref := strings.TrimPrefix(vv[1], "//")
+		// expected format is storeID@hash
+		parts := strings.SplitN(ref, "@", 2)
+		if len(parts) != 2 {
+			return nil, nil, nil, errors.Errorf("invalid oci-layout format '%s', must be oci-layout:///content-store@sha256:digest", vv[1])
+		}
+		st := llb.OCILayout(parts[0], parts[1],
```

Comment:
This needs to either call `ResolveImageConfig` (currently not implemented on daemon side) or

pull in image config from metadata like the `input` option does couple of lines below. If using metadata then the client needs to pass the image config as well with this value what will get messy when resolving config for a specific platform so implementing `ResolveImageConfig` is preferred.

Currently when this is not implemented you will pull in layers from a local image but you will not pull in `ENV` that this image defined for example.

Reply:
When you add this update the Dockerfile test to also check that `ENV` is preserved.